### PR TITLE
Modernize app example

### PR DESCRIPTION
**Pull request by stevapple**
Update `LocalDebugging` example to reflect the latest SwiftUI practice.

### Motivation:

Since we already required Swift 5.5 with Concurrency (Xcode 13), it's natural to update the app part to reflect latest SwiftUI changes, which makes the code neater and more expressive.

### Modifications:

- Update `LocalDebugging/MyApp` to use SwiftUI lifecycle;
- Update `LocalDebugging/MyApp` to reflect SwiftUI best practice;
- Update `LocalDebugging/Shared` to use Swift 5.5.

### Result:

A better demo app.

### CODE REVIEW

**Comment by tomerd** [comment]
thank @stevapple !

lets get #228 in first

**Comment by tomerd** [comment]
@swift-server-bot test this please

File: Examples/LocalDebugging/MyApp/MyApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist
Comment (tomerd) [inline Examples/LocalDebugging/MyApp/MyApp.xcodeproj/project.xcworkspace/xcshareddata/IDEWorkspaceChecks.plist:1]: is this file necessary?

File: Examples/LocalDebugging/MyApp/MyApp.xcodeproj/project.xcworkspace/contents.xcworkspacedata
Comment (tomerd) [inline Examples/LocalDebugging/MyApp/MyApp.xcodeproj/project.xcworkspace/contents.xcworkspacedata:1]: is this file necessary?

Comment (stevapple) [reply Examples/LocalDebugging/MyApp/MyApp.xcodeproj/project.xcworkspace/contents.xcworkspacedata:1]: I think it’s not 👌 will remove

**Comment by stevapple** [comment]
I've rebased the branch against #228, so there should be no pain to merge.

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by tomerd** [comment]
@swift-server-bot test this please

File: Examples/LocalDebugging/MyApp/MyApp/ContentView.swift
Comment (stevapple) [reply Examples/LocalDebugging/MyApp/MyApp/ContentView.swift:70]: Uh yes.

I changed it into `() async -> String` because we need to ensure the label is changed in every path.  `throws` could be preserved here and it doesn't affect.

Comment (fabianfett) [reply Examples/LocalDebugging/MyApp/MyApp/ContentView.swift:70]: I would prefer something like this:
```swift
func register() await -> String {
    do {
        return self.runNetworkRegister(name: self.name, password: self.password)
    } catch as CommunicationError {
        return error.reason
    } catch {
        return error.localizedDescription
    }
}
```

Comment (stevapple) [reply Examples/LocalDebugging/MyApp/MyApp/ContentView.swift:70]: I would rather suggest we `catch` in the button action? The `register() [async] [throws]-> String` API is expected not to confuse an actual response and a thrown `Error`. `runNetworkRegister(name:password:)` looks redundant here.

Comment (stevapple) [reply Examples/LocalDebugging/MyApp/MyApp/ContentView.swift:70]: Conditional `catch` can also be eliminated here by implementing `CustomStringConvertible`.

**Review by fabianfett — APPROVED** [review]
Let's wait for @tomerd approval!

**Comment by fabianfett** [comment]
@tomerd ping

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-server-bot test this please

**Comment by stevapple** [comment]
ping @tomerd @fabianfett

**Pull request merged**
